astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -825,9 +831,10 @@ def test_strip_line_trailing_whitespace(parallel, read_basic):
     row.
     """
     text = 'a b c\n1 2 \n3 4 5'
-    with pytest.raises(CParserError) as e:
+    with pytest.raises(InconsistentTableError) as e:
         ascii.read(StringIO(text), format='fast_basic', guess=False)
-    assert 'not enough columns found in line 1' in str(e)
+    assert 'header columns (3) inconsistent with data columns in data line 0' \
+        in str(e)
 
```

Comment:
API change: `ascii.read()` with `fast_reader` raises `InconsistentTableError` instead of `CParserError` (i.e. more in line with non-fast reader) when no. of data columns does not match header columns.